sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -1342,6 +1345,52 @@ abstract class DynamicPartitionPruningSuiteBase
       }
     }
   }
+
+  test("SPARK-32748: propagate local properties to dynamic pruning thread") {
+    def checkPropertyValueByUdfResult(propKey: String, propValue: String): Unit = {
+      spark.sparkContext.setLocalProperty(propKey, propValue)
+      val df = sql(
+        s"""
+           |SELECT compare_property_value(f.date_id, '$propKey', '$propValue') as col
```

Comment:
Is `compare_property_value` really evaluated in `SubqueryBroadcastExec` threads?

Reply:
Yes, IIUC udf created by `spark.udf.register` is not foldable, so it will be evaluated in tasks. Besides, otherwise TaskContext.get() will get null if it's not run in task I think.